src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -70,6 +71,17 @@ public static FormulaValue FirstN(IRContext irContext, FormulaValue[] args)
             var arg0 = (TableValue)args[0];
             var arg1 = (NumberValue)args[1];
 
+            if (arg0 is QueryableTableValue queryableTable)
+            {
+                try
+                {
+                    return queryableTable.FirstN((int)arg1.Value);
+                }
+                catch (NotDelegableException)
+                {
+                }
+            }
+
             var rows = arg0.Rows.Take((int)arg1.Value);
```

Comment:
>arg0.Rows.Take((int)arg1.Value); [](http://example.com/codeflow?start=23&length=32)

For a future PR ...  (@anderson-joyle )
The symmetry between line 78 and 85 makes me wonder ...
What if FirstN was a virtual on TableValue, and the default impl was line 85. 

Then this method becomes trivial....